docker/ci/build

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+set -eux
+
+BUNDLER_PATH=.bundle/gems
+
+function build_and_test_image() {
+  container_name=$1
+
+  : ===
+  : === run linter on the docker files
+  : ===
+   bundle exec puppet-docker local-lint $container_name
+
+  : ===
+  : === build and test $container_name
+  : ===
+  bundle exec puppet-docker build $container_name --repository pcr-internal.puppet.net/release-engineering
+  bundle exec puppet-docker spec $container_name
+}
+
+function push_image() {
+  container_name=$1
+  : ===
+  : === push $container_name
+  : ===
+  bundle exec puppet-docker push $container_name --repository pcr-internal.puppet.net/release-engineering
+}
+
+: ===
+: === bundle install to get ready
+: ===
+bundle install --path $BUNDLER_PATH
+
+: ===
+: === pull updated base images
+: ===
+bundle exec puppet-docker update-base-images ubuntu:16.04 postgres:9.6.8
+
+container_list=(puppetdb puppetdb-postgres)
+
+: ===
+: === build and test all the images before we push anything
+: ===
+for container in ${container_list[@]}; do
```

Comment:
I believe quoting this will do the same per-item expansion/quoting as for "$@": `"${container_list[@]}"`